packages/react/src/components/TabContent/TabContent.js

```diff
@@ -6,24 +6,44 @@
  */
 
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import classNames from 'classnames';
 import { settings } from 'carbon-components';
+import { selectorTabbable } from '../../internal/keyboard/navigation';
 
 const { prefix } = settings;
 
+/**
+ * Determine if the node within the provided ref contains content that is tabbable.
+ */
+function useTabbableContent() {
+  const [hasTabbableContent, setHasTabbableContent] = useState(false);
+
+  const ref = useCallback((node) => {
+    if (node !== null) {
+      // we don't need to know which element or how many are tabbable, just that there is _something_ tabbable
+      setHasTabbableContent(node.querySelector(selectorTabbable));
+    }
+  }, []);
+
+  return [hasTabbableContent, ref];
+}
+
 const TabContent = (props) => {
   const { className, selected, children, ...other } = props;
   const tabContentClasses = classNames(`${prefix}--tab-content`, {
     [className]: className,
   });
+  const [hasTabbableContent, ref] = useTabbableContent();
```

Comment:
Love this callback pattern, seems super clever for running code when a component "mounts". I wonder if there would there be an easy way to have this hook accept a ref just to make composition with hooks that use refs a little easier?

Something like:

```js
function useTabbableContent(ref) {
  const [hasTabbableContent, setHasTabbableContent] = useState(false);

  // useEffect on the server, useLayoutEffect on the client
  useIsomorphicEffect(() => {
    if (ref.current) {
      setHasTabbableContent(ref.current.querySelector(selectorTabbable));
    }
  });

  return hasTabbableContent;
}
```

I _think_ these would run at similar times but wasn't sure 🤷

Reply:
Great idea! Updated